### PR TITLE
Switching from Flash to JS for copy to clipboard. Issue #1072

### PR DESCRIPTION
**Pull request by mradziwon**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -7,5 +7,6 @@
 //= require jquery
 //= require jquery_ujs
 //= require jquery.color
-//= require ZeroClipboard
+//= require clipboard.min
+//= require clipboard
```
Comment (simi) [inline app/assets/javascripts/application.js]: Why both versions are required?

Comment (mradziwon) [inline app/assets/javascripts/application.js]: Thanks for advice I can change it to gem.

**Comment by simi** [comment]
Isn't the `de.yml ` added by accident?

**Comment by mradziwon** [comment]
Yes it is sorry.

**Comment by nateberkopec** [comment]
Code LGTM but I think this can be squashed into a single commit.

**Comment by simi** [comment]
LGTM here now also. Squash will be awesome.

**Comment by nateberkopec** [comment]
Maintainers should double check on the behavior for Safari and determine if it's acceptable. Clipboard.js is supposed to have some graceful degradation.

screen shot 2016-01-10 at 9 17 57 am

**Comment by simi** [comment]
@nateberkopec that's the known thing and for Safari, there's a fallback behaviour selecting text via Selection API and tooltip to press `cmd/ctrl + c`. Maybe better text should be used.

copy

File: app/assets/javascripts/copy-to-clipboard.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      copyButton.removeClass('clipboard-is-active');
+    }, 1000);
+  }
+
+  copyToClipboard.on('success', function(event) {
+    showNotification(event);
+  });
+
+  copyToClipboard.on('error', function(event) {
+    $('.gem__code__tooltip--copied').text('ctrl/cmd + c')
```
Comment (simi) [inline app/assets/javascripts/copy-to-clipboard.js]: missing semicolon

**Comment by mradziwon** [comment]
@simi Any suggestions for better text? I think this one is clear and short. I squashed it to one commit as you suggested.

**Comment by simi** [comment]
Something like "Press Ctrl/Cmd+C to copy.". But that will need translation also. On the other side, when browser is not supported by clipboard.js, maybe it will be enough to hide that copy button. Let's wait for some feedback.

**Comment by dwradcliffe** [comment]
Trying to remember, didn't we decide to do what github does for fallback?

On Sunday, January 10, 2016, Josef Šimánek <notifications@github.com> wrote:

> Something like "Press Ctrl/Cmd+C to copy.". But that will need translation
> also. On the other side, when browser is not supported by clipboard.js,
> maybe it will be enough to hide that copy button. Let's wait for some
> feedback.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubygems/rubygems.org/pull/1170#issuecomment-170366052>
> .
>


-- 
---
David Radcliffe

**Comment by simi** [comment]
AFAIK GH uses ZeroClipboard and fallbacks to swf.

**Comment by knappe** [comment]
@mradziwon I believe this PR can be closed.  We've been trying to move away from using flash and going to clipboard.js for a while.  Looks like that finally happened here: https://github.com/rubygems/rubygems.org/commit/a900e402c16ec80ef7b12d7907cbc13787f12214

**Pull request closed without merging**